src/urllib3/response.py

```diff
@@ -148,6 +153,24 @@ def flush(self) -> bytes:
             return b""
 
 
+if zstd is not None:
+
+    class ZstdDecoder(ContentDecoder):
+        def __init__(self) -> None:
+            self._obj = zstd.ZstdDecompressor().decompressobj()
+
+        def decompress(self, data: bytes) -> bytes:
+            if not data:
+                return b""
+            return self._obj.decompress(data)  # type: ignore[no-any-return]
+
+        def flush(self) -> bytes:
+            ret = self._obj.flush()
+            if not self._obj.eof:
+                raise DecodeError("Invalid zstandard data")
```

Comment:
Strictly speaking, the message can be this: zstd data ends in an incomplete frame.

Reply:
It's a little long, how about "zstd data is incomplete"?